nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,156 @@
+                'time_downsamplers': ['mean'],
+                'value_downsamplers': ['min', 'mean', 'max'],
+                'minimum': 0.0,
+            },
+        })
+
+        def get_module_name(self):
+            return 'monitor.http.survey'
+
+    class SurveyInfoStreamsData(object):
+        def __init__(self, node, frequency, channel, snr, neighbors):
+            self.node = node
+            self.frequency = frequency  # either 2 or 5, corresponding to 2.4GHz and 5GHz
+            self.channel = channel
+            self.snr = snr
+            self.neighbors = neighbors
+            self.neighbor_count = len(self.neighbors)
+
+    ds_pool.register(SurveyInfoStreamsData, SurveyInfoStreams)
+
+    DATASTREAM_SUPPORTED = True
+except ImportError:
+    pass
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Stores node's reported channel, SNR and neighbor data into the monitoring schema. Will
+    only run if HTTP monitor module has previously fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+        # Should I go through the existing clients as well?
+        # existing_clients = {}
+        #for client in node.monitoring.network.clients():
+        #    existing_clients[client.client_id] = client
+
+        version = context.http.get_module_version("core.clients")
+        if version == 0:
+            # Unsupported version or data fetch failed (v0)
+            return context
+
+        channel = self.getNodeChannel(context, node)
+        snr = self.getNodeSNR(context, node)
+        neighbors = self.getNodeNeighbors(context, node)
+        print(neighbors)
+        if DATASTREAM_SUPPORTED:
+            # Store client count into datastream.
+            context.datastream.monitor_http_clients = SurveyInfoStreamsData(node, "2", channel, snr, neighbors)
+
+        return context
+
+    def getNodeChannel(self, context, node, frequency_band="2"):
+        """
+        Extracts the channel number from the context for the specified frequency band.
+        :param context: Current context
+        :param node: Node that is being processed
+        :param frequency_band: either "2" or "5", corresponding to 2.4GHz and 5GHz.
+        :return: channel currently assigned to the node at the specified frequency band.
+        """
+        try:
+            for interface in context.http.core.wireless.interfaces:
+                if str(context.http.core.wireless.interfaces[interface].frequency)[:1] == frequency_band:
+                    return context.http.core.wireless.interfaces[interface].channel
+        except NameError:
+            print("Error parsing JSON file for " + node)
```

Comment:
Do not use `print` for logging, use `self.logger` instead, eg. `self.logger.warning("Error parsing JSON ...")`.